src/pip/_internal/commands/uninstall.py

```diff
@@ -59,6 +61,17 @@ def run(self, options, args):
 
         reqs_to_uninstall = {}
         for name in args:
+            if is_archive_file(name):
```

Comment:
`is_archive_file` only check if `name` ends with a known archive extension:
https://github.com/pypa/pip/blob/00e531a16e3079249b60d959328449fb1988ea8d/src/pip/_internal/req/constructors.py#L48-L54

But `myfile.zip` is a valid python distribution name and [swh.loader.tar](https://pypi.org/project/swh.loader.tar/) even exists (and `is_archive_file('swh.loader.tar')` returns True).

You'll likely have to take inspiration from https://github.com/pypa/pip/blob/00e531a16e3079249b60d959328449fb1988ea8d/src/pip/_internal/req/constructors.py#L300-L388 (or maybe use it ?) to mimic pip interpretation of arguments.

Reply:
Thanks! I did not notice this edge case. :sweat_smile:
I believe I fixed the bug, I'll rebase from master (squash commits) after you'll mark the conversation as resolved.
By the way, currently the functionality of installing `swh.loader.tar`-like packages is failing on my testing environments (maybe a bug? I'll consider opening an issue):
1. Windows 10 Pro (10.0.19041) + python 3.9.0 + pip 20.2.4
2. Ubuntu 19.04 + python 3.7.3 + pip 18.1

Reply:
Thanks @xavfernandez ! I did not notice this edge case. 😅
I believe I fixed the bug, I'll rebase from master (squash commits) after you'll mark the conversation as resolved.